x/sequencer/keeper/get_and_set.go

```diff
@@ -17,6 +17,9 @@ import (
 // SetSequencer : write to store indexed by address, and also by status
 // Note: do not call with sentinel sequencer
 func (k Keeper) SetSequencer(ctx sdk.Context, seq types.Sequencer) {
+	if seq.Sentinel() {
+		panic("set sentinel proposer")
+	}
```

Comment:
I really feel this is a good thing to have

Reply:
problem is that it can be called in an end blocker (like from gov fraud proposal) which will halt the chain.
not sure about it. why not just return an err?

Reply:
We could of course, but it really complicates the API. Also, I'm not sure how I'd handle the error.

Everything is horribly wrong if this error occurs. Also, I think we can class invariant breaking errors in two

1. It will be ok if the hub continues progressing while the invariant is broken. We will be able to fix it later
2. Once the invariant is broken, the behaviour can become more and more corrupted over time

I think this is case 2. I really don't want to be in a situation where this happened, but we didn't notice. In principle we'd have to recover the hub somehow, and fix corrupted state, but it could have been corrupted for a long time already.

I do think panics are warranted tbh

Reply:
problem is - does it makes sense to take down an entire hub which serves hundreds of rollapps for a corruption in one? 

also not sure the implications are very tragic. as long as it couldn't lead to direct loss of funds not sure it's worth it.

also in the playground we had it now and it didn't seem like it caused any tragic non-recoverable state (also happened just after hard fork which is an edge case). 

if we had this code now, do you think it would worth to halt the playground for a hard-fork gone wrong? I'm not sure.

Reply:
> problem is - does it makes sense to take down an entire hub which serves hundreds of rollapps for a corruption in one?

It's not one though, since rollapps share the sentinel sequencer

Reply:
IMO it's like out of bounds access in c++ or something, it's really hard to say what could go wrong